test/compound/TestSupplyVault.t.sol

```diff
@@ -333,26 +333,45 @@ contract TestSupplyVault is TestSetupVaults {
         assertApproxEqAbs(rewardsAmount2, rewardsAmount3, 1e9, "unexpected rewards amount 2-3"); // not exact because of compounded interests
     }
 
-    function testNotOwnerShouldNotTransferTokens(uint256 _amount) public {
+    function testNotOwnerShouldNotSetRecipient() public {
         vm.prank(address(1));
         vm.expectRevert("Ownable: caller is not the owner");
-        daiSupplyVault.transferTokens($token, address(2), _amount);
+        daiSupplyVault.setRewardsRecipient(address(1));
     }
 
-    function testOwnerShouldTransferTokens(
-        address _to,
-        uint256 _deal,
-        uint256 _toTransfer
-    ) public {
-        vm.assume(_to != address(daiSupplyVault));
-        _toTransfer = bound(_toTransfer, 0, _deal);
-        deal($token, address(daiSupplyVault), _deal);
+    function testOwnerShouldSetRecipient() public {
+        daiSupplyVault.setRewardsRecipient(address(1));
+        assertEq(daiSupplyVault.recipient(), address(1));
+    }
+
+    function testCannotSetRecipientToZeroAddress() public {
+        vm.expectRevert(abi.encodeWithSelector(SupplyVaultBase.ZeroAddress.selector));
+        daiSupplyVault.setRewardsRecipient(address(0));
+    }
+
+    function testCannotTransferRewardsToNotSetZeroAddress() public {
+        vm.expectRevert(abi.encodeWithSelector(SupplyVaultBase.ZeroAddress.selector));
+        daiSupplyVault.transferRewards();
+    }
+
+    function testEverybodyCanTransferRewardsToRecipient(uint256 _amount) public {
+        vm.assume(_amount > 0);
+
+        daiSupplyVault.setRewardsRecipient(address(1));
+        assertEq(ERC20(MORPHO_TOKEN).balanceOf(address(1)), 0);
```

Comment:
testing that address 1 didn't farm any MORPHO on the mainnet ? 😅

Reply:
this could be the case any other address right?